internal/controller/biossettings_controller.go

```diff
@@ -984,410 +885,288 @@ func (r *BIOSSettingsReconciler) applyBiosSettingOnServer(
 			diff[key] = value
 		}
 	}
 
-	if len(diff) > 0 {
-		log.V(1).Info("current BIOS settings on the server",
-			"currentSettings", currentSettings,
-			"currentSettingsKeys", keys,
-			"current Spec settings", currentPrioritySettings,
-			"difference", diff)
-	}
-
 	return diff, errors.Join(errs...)
 }
 
-func (r *BIOSSettingsReconciler) getBIOSVersionAndSettingDifference(
-	ctx context.Context,
-	log logr.Logger,
-	bmcClient bmc.BMC,
-	biosSettings *metalv1alpha1.BIOSSettings,
-	server *metalv1alpha1.Server,
-) (currentbiosVersion string, diff redfish.SettingsAttributes, err error) {
-
+func (r *BIOSSettingsReconciler) getBIOSVersionAndSettingsDiff(ctx context.Context, log logr.Logger, bmcClient bmc.BMC, settings *metalv1alpha1.BIOSSettings, server *metalv1alpha1.Server) (string, redfish.SettingsAttributes, error) {
 	completeSettings := make(map[string]string)
-	for _, settings := range biosSettings.Spec.SettingsFlow {
-		for key, value := range settings.Settings {
+	for _, flowItem := range settings.Spec.SettingsFlow {
+		for key, value := range flowItem.Settings {
 			completeSettings[key] = value
 		}
 	}
 
-	diff, err = r.getCurrentSettingDifference(ctx, log, bmcClient, completeSettings, server)
-
+	diff, err := r.getSettingsDiff(ctx, bmcClient, completeSettings, server)
 	if err != nil {
-		log.V(1).Info("Failed to get bios setting differences for some settings", "error", err)
-		return "", diff, fmt.Errorf("failed to find diff for some bios settings: %v", err)
+		return "", diff, fmt.Errorf("failed to find BIOS settings difference: %w", err)
+	}
+	if len(diff) > 0 {
+		log.V(1).Info("Found BIOSSettings difference on Server", "Server", server.Name, "SettingsDifference", diff)
 	}
 
-	// fetch the current bios version from the server bmc
-	currentBiosVersion, err := bmcClient.GetBiosVersion(ctx, server.Spec.SystemURI)
+	version, err := bmcClient.GetBiosVersion(ctx, server.Spec.SystemURI)
 	if err != nil {
-		return "", diff, fmt.Errorf("failed to load bios version: %w for server %v", err, server.Name)
+		return "", diff, fmt.Errorf("failed get BIOS version: %w", err)
 	}
 
-	return currentBiosVersion, diff, nil
+	return version, diff, nil
 }
 
-func (r *BIOSSettingsReconciler) checkForRequiredPowerStatus(
-	server *metalv1alpha1.Server,
-	powerState metalv1alpha1.ServerPowerState,
-) bool {
-	return server.Status.PowerState == powerState
+func (r *BIOSSettingsReconciler) isServerInPowerState(server *metalv1alpha1.Server, state metalv1alpha1.ServerPowerState) bool {
+	return server.Status.PowerState == state
 }
 
-func (r *BIOSSettingsReconciler) checkIfMaintenanceGranted(
-	log logr.Logger,
-	biosSettings *metalv1alpha1.BIOSSettings,
-	server *metalv1alpha1.Server,
-) bool {
-	if biosSettings.Spec.ServerMaintenanceRef == nil {
+func (r *BIOSSettingsReconciler) isServerInMaintenance(log logr.Logger, settings *metalv1alpha1.BIOSSettings, server *metalv1alpha1.Server) bool {
+	if settings.Spec.ServerMaintenanceRef == nil {
 		return true
```

Comment:
Assuming that the absence of a maintenance ref means that the server is in maintenance looks suspicious.